include/bluetooth/gatt.h

```diff
@@ -1208,10 +1208,30 @@ struct bt_gatt_exchange_params {
  *
  *  @note Shall only be used once per connection.
  *
+ *  The Response comes in callback @p params->func. The callback is run from
+ *  the Bluetooth event context. @p *params must remain valid until start of
+ *  callback.
+ *
+ *  This function will block while the ATT request queue is full, except when
+ *  called from Bluetooth event context. When called from Bluetooth context,
+ *  this function will instead instead return `-ENOMEM` if it would block to
+ *  avoid a deadlock.
```

Comment:
It's not bad to have, but it does seem a bit superfluous to describe the return value both here and in the `@retval`